modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorFieldMapperTests.java

```diff
@@ -146,7 +146,7 @@ public void init() throws Exception {
         mapperService = indexService.mapperService();
 
         String mapper = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("doc")
-            .startObject("_field_names").field("enabled", false).endObject() // makes testing easier
+            .startObject("_field_names").endObject() // makes testing easier
```

Comment:
+1, we should disallow `_field_names` entirely for newly created indices.

Reply:
I'm not really following you here, #27239 talks about disabeling the `enable`option. What would disallowing the whole field mean? Do we still create the field (since its default is enabled:true?) What's the whole breaking/deprecation story around this then?

Reply:
Sorry I wasn't clear I meant that we should disallow setting `_field_names`  entirely since we're on master but it seems that the scope of this pr is only the deprecation. However I agree with Julie that we should try to avoid mentioning `_field_names` in this test since this shouldn't affect the behavior of the percolator. If it turns too complicated we can leave this for a follow up but we'll need to do it anyway since the goal is to disallow disallowing the `_field_names` entirely and this test relies on this forbidden behavior.